File: Lab_02/APPOO2_3/src/com/company/CustomList.java
package com.company;

/**
 * Created by bossa on 13.06.2017.
 */
import java.util.*;

class CustomList {
    List<SmartPhone> list;

    CustomList()
    {
        this.list = new ArrayList<>();
    }

    void addElement(SmartPhone p)
    {
        this.list.add(p);
    }

    List<SmartPhone> getList()
    {
        return list;
    }

    List<SmartPhone> findByBrand(String brand)
    {
        List<SmartPhone> temp = new ArrayList<>();
        for (SmartPhone t : list) {
            if (t.getBrand().equals(brand)) {
                temp.add(t);
            }
        }
        return temp;
    }

    CustomDeque OutputBrandsByCondition(String brand){
        List<SmartPhone> result = this.findByBrand(brand);
        CustomDeque temp = new CustomDeque();
        if(result.size()!=0){
            System.out.println("Found:");
            for(SmartPhone obj : result){
                System.out.println(obj);
                temp.addElement(obj);
                this.list.remove(obj);
            }
        }
        else{
            System.out.println("Nothing Found!!!");
        }
        return temp;
    }

    void sortDescending(){
        this.list.sort(new DescendingComparator());
    }

    void sortAscending(){
        this.list.sort(new AscedingComparator());
    }

    void PrintList(){
        System.out.println("Print List:");
        for(SmartPhone obj: list){
            System.out.println(obj);
        }
    }

    int CountByPrice(int price){
        int result = 0;
        for(SmartPhone obj: list){
            if(obj.getPrice() > price){
                result += 1;
            }
        }
        return result;
    }

    void writeToDeque(Deque<SmartPhone> dq){
        if(!dq.isEmpty())dq.clear();
        dq.addAll(list);
    }

}

File: Lab_03/appoo3/appoo3/CustomList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace appoo3
{
    class CustomList
    {
        private List<SmartPhone> list;

        public CustomList()
        {
            this.list = new List<SmartPhone>();
        }

        public List<SmartPhone> GetList()
        {
            return list;
        }

        public void AddElement(SmartPhone obj)
        {
            this.list.Add(obj);
        }

        public List<SmartPhone> FindByBrand(String brand)
        {
            List<SmartPhone> temp = new List<SmartPhone>();
            foreach (var obj in list)
            {
                if (obj.GetBrand().Equals(brand))
                {
                    temp.Add(obj);
                }
            }
            return temp;
        }

        public CustomDeque OutputToDeque(String brand)
        {
            List<SmartPhone> temp = FindByBrand(brand);
            CustomDeque result = new CustomDeque();
            if (temp.Count() != 0)
            {
                foreach (var obj in temp)
                {
                    result.AddElement(obj);
                    this.list.Remove(obj);
                }
            }
            else
            {
                Console.WriteLine("Nothing Found!");
            }
            return result;
        }

        public void SortDescending()
        {
            list.Sort(new DescendingComparator());
        }

        public void SortAscending()
        {
            list.Sort(new AscendingComparator());
        }

        public int CountByPrice(int price)
        {
            int count = 0;
            foreach (var obj in list)
            {
                if (obj.GetPrice() > price)
                {
                    count += 1;
                }
            }
            return count;
        }

        public void Print()
        {
            Console.WriteLine("\n[Print List]\n");
            foreach (var obj in list)
            {
                Console.WriteLine(obj);
            }
        }

        public void OutputByBrand(String brand)
        {
            if (list.Count() != 0)
            {
                Console.WriteLine("Found:");
                foreach (var obj in list)
                {
                    if (obj.GetBrand().Equals(brand))
                    {
                        Console.WriteLine(obj);
                    }
                }
            }
            else
            {
                Console.WriteLine("Nothing Found:!");
            }
        }
    }
}
